lib/Paneset/PaneResizeContainer.js

```diff
@@ -1,30 +1,100 @@
 import React, { useState, useRef, useEffect } from 'react';
 import cloneDeep from 'lodash/cloneDeep';
 import debounce from 'lodash/debounce';
+import isEqual from 'lodash/isEqual';
 import insertByClientRect from './insertByClientRect';
 import PaneResizeHandle from './PaneResizeHandle';
 import PaneResizeCursor from './PaneResizeCursor';
 import { ResizeProvider } from './ResizeContext';
 import css from './PaneResize.css';
 
-const PaneResizeContainer = ({ isRoot, children, onElementResize }) => {
+const PaneResizeContainer = ({ isRoot, children, onElementResize, parentElement }) => {
   const [dragging, setDragging] = useState(false);
   const [cursorX, setCursorX] = useState(0);
   const [handles, setHandles] = useState([]);
   const [activeHandle, setActiveHandle] = useState(null);
   const [update, setUpdate] = useState(0); // eslint-disable-line no-unused-vars
+  const [blocking, setBlocking] = useState(null);
   const container = useRef();
   const handleData = useRef([]);
   const containerRect = useRef();
   const activeRef = useRef();
-  const debouncedUpdate = useRef(debounce(
-    () => {
-      containerRect.current = container.current?.getBoundingClientRect();
-      setUpdate(u => u + 1);
-    },
-    300,
-    { trailing: true }
-  )).current;
+  const observer = useRef();
+
+  const observerCallback = useRef(
+    (mutationList) => {
+      setBlocking(curBlocker => {
+        let blocker = curBlocker;
+        mutationList.forEach((m) => {
+          if (m.type === 'childList') {
+            if (m.addedNodes.length > 0) {
+              // check against full container bounds
+              m.addedNodes.forEach((n) => {

```

Comment:
Would it be just enough to stop when at least the one is found? It seems like in the current situation only the last (in case if there are multiple full-width ones) will be stored.

```suggestion
              m.addedNodes.some((n) => {
```

Reply:
Cool.. I've typically used `.some` for it's boolean return, but I suppose it is like a self-terminating forEach.

Reply:
Yeah, but in this case, it is needed to return some bool in order for it to understand whether it needs to continue. Maybe the easier way would be to use `.find` which will return the first matched entry.

Reply:
So there was another slight gotcha - The `addedNodes` and `removedNodes` are NodeLists... had to use `Array.from()`.